nncf/quantization_metric.py

```diff
@@ -0,0 +1,241 @@
+            for column_name in self.header:
+                drow[column_name] = self.stat[column_name][p]
+            drow[self.PARAMS_STR] = p
+            row = [drow[h] for h in self.header]
+            data.append(row)
+        data[0] =[self.PARAMS_STR] + header_for_table
+        table.add_rows(data)
+        return table
+
+class MemoryСostMetric(BaseMetric):
+    PARAMS_STR = 'params'
+    NAME_STR = 'MemoryCost'
+
+    EXPECTED_IMPROVEMENT_COST_MEMORY_STR = 'expected improvement cost memory'
+    MAX_TENSOR_ACTIVATIONS_SHAPE_STR = 'max tensor activations shape'
+    COST_MEMORY_FP_WEIGHTS_STR = 'cost memory fp weights'
+    COST_MEMORY_COMPRESSED_WEIGHTS_STR = 'cost memory compressed weights'
+
+    def __init__(self, compressed_model: NNCFNetwork):
+        super().__init__()
+        self._compressed_model = compressed_model
+        self.header = [self.EXPECTED_IMPROVEMENT_COST_MEMORY_STR]
+        self.stat = {}
+
+    def collect(self):
+        self.stat[self.COST_MEMORY_FP_WEIGHTS_STR] = 0
+        self.stat[self.COST_MEMORY_COMPRESSED_WEIGHTS_STR] = 0
+        fp_num_bits = 32
+        nncf_modules = self._compressed_model.get_nncf_modules()
+
+        for nncf_module in nncf_modules.values():
+            count_el = np.prod(nncf_module.weight.shape)
+            self.stat[self.COST_MEMORY_FP_WEIGHTS_STR] += count_el * fp_num_bits
+            if len(nncf_module.pre_ops) > 0:
+                # pylint: disable=protected-access
+                num_bits = nncf_module.pre_ops._modules['0'].op.num_bits
+                self.stat[self.COST_MEMORY_COMPRESSED_WEIGHTS_STR] += count_el * num_bits
+            else:
+                self.stat[self.COST_MEMORY_COMPRESSED_WEIGHTS_STR] += count_el * fp_num_bits
+        self.stat[self.EXPECTED_IMPROVEMENT_COST_MEMORY_STR] = self.stat[self.COST_MEMORY_FP_WEIGHTS_STR] /\
+             self.stat[self.COST_MEMORY_COMPRESSED_WEIGHTS_STR]
+
+        self.stat[self.MAX_TENSOR_ACTIVATIONS_SHAPE_STR] = max(self._compressed_model.input_infos,\
+             key=lambda k: np.prod(k.shape)).shape
+
+    def get_metric_table(self):
+        table = Texttable()
+        data = [self.header]
+        drow = {h: 0 for h in self.header}
+        for column_name in self.header:
+            drow[column_name] = self.stat[column_name]
+        row = [drow[h] for h in self.header]
+        data.append(row)
+        table.add_rows(data)
+        return table
+
+
+class ShareEdgesQuantizedDataPath(BaseMetric):
+    NAME_STR = 'ShareEdgesQuantizedDataPath'
+    COUNT_QUANTIZED_EDGES_STR = '% edges of the quantized data path'
+    QUANTIZED_EDGES_ATTR = 'quantized'
+
+    def __init__(self, compressed_model: NNCFNetwork):
+        super().__init__()
+        self._compressed_model = compressed_model
+        self._compressed_model.rebuild_graph()
+        self._compressed_graph = self._compressed_model.get_graph()
+        nx.set_edge_attributes(self._compressed_graph._nx_graph, False, self.QUANTIZED_EDGES_ATTR)
```

Comment:
I hope that this is a copy of the model's compressed graph you are working on, not the original compressed graph.